internal/controller/datadogagent/profile.go

```diff
@@ -77,13 +76,10 @@ func (r *Reconciler) computeProfileMerge(ddai *v1alpha1.DatadogAgentInternal, pr
 	// Copy the original DDAI and apply profile spec to create a fake "DDAI" to merge
 	profileDDAI := ddai.DeepCopy()
 	baseDDAI := ddai.DeepCopy()
-	if !agentprofile.IsDefaultProfile(profile.Namespace, profile.Name) {
-		// Clear owner reference to tie GC to the profile
-		baseDDAI.OwnerReferences = []metav1.OwnerReference{}
-	}
```

Comment:
Removed this since we removed tie between profile and DDAIs in https://github.com/DataDog/datadog-operator/pull/2001